src/main/java/org/broadinstitute/hellbender/utils/smithwaterman/SmithWatermanJavaAligner.java

```diff
@@ -74,24 +76,38 @@ public SmithWatermanAlignment align(final byte[] reference, final byte[] alterna
 
         final SmithWatermanAlignment alignmentResult;
 
+        //if exact match
         if (matchIndex != -1) {
             // generate the alignment result when the substring search was successful
-            final List<CigarElement> lce = new ArrayList<>(alternate.length);
-            lce.add(makeElement(State.MATCH, alternate.length));
+            final List<CigarElement> lce = Collections.singletonList(makeElement(State.MATCH, alternate.length));
             alignmentResult = new SWPairwiseAlignmentResult(AlignmentUtils.consolidateCigar(new Cigar(lce)), matchIndex);
         }
         else {
-            // run full Smith-Waterman
-            final int n = reference.length+1;
-            final int m = alternate.length+1;
-            final int[][] sw = new int[n][m];
-            final int[][] btrack=new int[n][m];
-
-            calculateMatrix(reference, alternate, sw, btrack, overhangStrategy, parameters);
-            alignmentResult = calculateCigar(sw, btrack, overhangStrategy); // length of the segment (continuous matches, insertions or deletions)
+            int matchIndex2 = -1;
```

Comment:
It's worth it to make these variable names more explicit at the cost of verbosity and call the first index `exactMatchIndex` and the second `singleMismatchIndex`.